### ISSUE TITLE
poetry-core 1.8.1: pytest fails on compile with `-Wint-conversion`

### ISSUE THREAD

**Issue by kloczek** [python-poetry/poetry#8914]
Upcoming gcc 14.x which snapshot is now available in fedora rawhide is more sensitive on some code issues.
```console
building 'extended.extended' extension
creating /home/tkloczko/rpmbuild/BUILD/poetry-core-1.8.1/tests/masonry/builders/fixtures/extended/build/temp.linux-x86_64-cpython-38
creating /home/tkloczko/rpmbuild/BUILD/poetry-core-1.8.1/tests/masonry/builders/fixtures/extended/build/temp.linux-x86_64-cpython-38/extended
/usr/bin/gcc -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -s
pecs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic
-fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -D_GNU_SOURCE -fPIC -fwrapv -ffat-lto-objects -fno-
semantic-interposition -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -spec
s=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fa
synchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -D_GNU_SOURCE -fPIC -fwrapv -ffat-lto-objects -fno-sem
antic-interposition -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-ha
rdened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -m64 -mtune=generic -fasynchronous-unwind-tables -fsta
ck-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -D_GNU_SOURCE -fPIC -fwrapv -ffat-lto-objects -fno-semantic-interposition -m64 -mtune
=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -O2 -g -grecord-gcc-switches -pipe -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -fno-omit-frame-pointer -mno-
omit-leaf-frame-pointer -fdata-sections -ffunction-sections -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -flto=auto -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,
-D_GLIBCXX_ASSERTIONS -Wall -Werror=format-security -fPIC -I/usr/include/python3.8 -c extended/extended.c -o /home/tkloczko/rpmbuild/BUILD/poetry-core-1.8.1/tests/masonry/builders/fixtures/
extended/build/temp.linux-x86_64-cpython-38/extended/extended.o
----------------------------- Captured stderr call -----------------------------
fatal: not a git repository (or any parent up to mount point /home/tkloczko)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
warning: build_py: byte-compiling is disabled, skipping.

extended/extended.c:13:9: error: initialization of ‘int’ from ‘void *’ makes integer from pointer without a cast [-Wint-conversion]
   13 |         NULL,
      |         ^~~~
extended/extended.c:13:9: note: (near initialization for ‘module_methods[0].ml_flags’)
error: command '/usr/bin/gcc' failed with exit code 1
```

BTW it would be good to enable issue tracker on https://github.com/python-poetry/poetry-core

**Comment by dimbleby** [comment]
merge requests are open on poetry-core, please go ahead!

**Comment by Secrus** [comment]
This has been fixed in python-poetry/poetry-core#671
